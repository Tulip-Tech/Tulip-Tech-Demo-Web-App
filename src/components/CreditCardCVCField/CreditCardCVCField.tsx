import React from 'react';

import TextField from '../TextField/TextField';

type Props = {
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
  onBlur?: React.FocusEventHandler<HTMLInputElement>;
  error?: string;
  value?: string;
};

const CreditCardCVCField: React.FC<Props> = ({ value, onChange, error, ...props }: Props) => {
  const formatCVC: React.ChangeEventHandler<HTMLInputElement> = (e) => {
    const clearValue = e.target.value.replace(/\D+/g, '');
    if (onChange) {
      e.target.value = clearValue.slice(0, 4);
      onChange(e);
    }
  };
  return (
    <TextField
      label={`CVC/CVV`}
      {...props}
      error={!!error}
      helperText={error ? error : null}
      name="cardCVC"
      type="text"
      value={value}
      onChange={formatCVC}
      pattern="\d*"
      placeholder="###"
      required
    />
  );
};

export default CreditCardCVCField;
