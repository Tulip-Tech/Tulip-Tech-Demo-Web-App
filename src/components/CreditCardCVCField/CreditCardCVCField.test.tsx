import React from 'react';
import { render } from '@testing-library/react';

import CreditCardCVCField from './CreditCardCVCField';

describe('<CreditCardCVCField>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<CreditCardCVCField value="" onChange={vi.fn()} />);

    expect(container).toMatchSnapshot();
  });
});
