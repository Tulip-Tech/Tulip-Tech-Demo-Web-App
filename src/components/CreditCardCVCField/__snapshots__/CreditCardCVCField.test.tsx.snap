// Vitest Snapshot v1

exports[`<CreditCardCVCField> > renders and matches snapshot 1`] = `
<div>
  <div
    class="_textField_e16c1b"
  >
    <label
      class="_label_e16c1b"
      for="text-field_1235_cardCVC"
    >
      CVC/CVV
    </label>
    <div
      class="_container_e16c1b"
    >
      <input
        class="_input_e16c1b"
        id="text-field_1235_cardCVC"
        name="cardCVC"
        pattern="\\\\d*"
        placeholder="###"
        required=""
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
